### PR TITLE
[CN-500] Rename Custom Class into User Code Deployment

### PR DESCRIPTION
**Pull request by mtyazici**

### CODE REVIEW

File: api/v1alpha1/hazelcast_types.go [hunk 1]
```diff
@@ -80,12 +80,12 @@ type HazelcastSpec struct {
...
 	// +optional
-	// +kubebuilder:default:={repository: "docker.io/hazelcast/platform-operator-agent", version: "0.1.5"}
+	// +kubebuilder:default:={repository: "docker.io/hazelcast/platform-operator-agent", version: "0.1.6"}
 	Agent *AgentConfiguration `json:"agent,omitempty"`
 
-	// Custom Classes to Download into Class Path
+	// User Codes to Download into CLASSPATH
 	// +optional
-	CustomClass *CustomClassConfiguration `json:"customClass,omitempty"`
+	UserCodeDeployment *UserCodeDeploymentConfig `json:"userCode,omitempty"`
```
Comment (SeriyBg) [inline api/v1alpha1/hazelcast_types.go:88]: Maybe it's better to stick with `userCodeDeployment` in the YAML for consistency?

Comment (mtyazici) [reply api/v1alpha1/hazelcast_types.go:88]: Sure, I will change it

**Review by dzeromski-hazelcast — APPROVED** [review]

File: api/v1alpha1/hazelcast_types.go [hunk 2]
```diff
@@ -286,13 +286,13 @@ func (c *ExposeExternallyConfiguration) IsEnabled() bool {
 	return c != nil && !(*c == (ExposeExternallyConfiguration{}))
 }
 
-// Returns true if customClass.bucketConfiguration is specified.
-func (c *CustomClassConfiguration) IsBucketEnabled() bool {
+// Returns true if userCode.bucketConfiguration is specified.
```
Comment (SeriyBg) [inline api/v1alpha1/hazelcast_types.go:289]:
```suggestion
// Returns true if userCodeDeployment.bucketConfiguration is specified.
```

File: api/v1alpha1/hazelcast_types.go [hunk 3]
```diff
@@ -286,13 +286,13 @@ func (c *ExposeExternallyConfiguration) IsEnabled() bool {
...
+func (c *UserCodeDeploymentConfig) IsBucketEnabled() bool {
 	return c != nil && c.BucketConfiguration != nil
 }
 
-// Returns true if customClass.configMaps configuration is specified.
-func (c *CustomClassConfiguration) IsConfigMapEnabled() bool {
+// Returns true if userCode.configMaps configuration is specified.
```
Comment (SeriyBg) [inline api/v1alpha1/hazelcast_types.go:294]:
```suggestion
// Returns true if userCodeDeployment.configMaps configuration is specified.
```

**Review by SeriyBg — APPROVED** [review]

**Pull request merged**